commit df41b42914892c539de4ec16a6d9184d031ae0bf
Author: Kamal Mostafa <kamal@whence.com>
Date:   Mon Oct 22 15:52:44 2012 -0700

    spi/spi-bcm2708: respect per-transfer SPI clock speed_hz value
    
    The bcm2708 SPI driver's bcm2708_process_transfer() was ignoring the
    per-transfer speed_hz value even when it was provided (it always just
    used the spi device's max_speed_hz value).  Now, per-transfer speed_hz
    values are respected.
    
    Also added debug print to bcm2708_setup_state() to help keep an eye on
    the configured SPI parameters.
    
    Signed-off-by: Kamal Mostafa <kamal@whence.com>

Index: linux-3.2.46/drivers/spi/spi-bcm2708.c
===================================================================
--- linux-3.2.46.orig/drivers/spi/spi-bcm2708.c	2013-07-26 19:33:55.000000000 +0000
+++ linux-3.2.46/drivers/spi/spi-bcm2708.c	2013-07-26 19:37:20.000000000 +0000
@@ -259,6 +259,10 @@
 	if (state) {
 		state->cs = cs;
 		state->cdiv = cdiv;
+		dev_dbg(dev, "setup: want %d Hz; "
+			"bus_hz=%lu / cdiv=%u == %lu Hz; "
+			"mode %u: cs 0x%08X\n",
+			hz, bus_hz, cdiv, bus_hz/cdiv, mode, cs);
 	}
 
 	return 0;
@@ -277,7 +281,8 @@
 
 	if (xfer->bits_per_word || xfer->speed_hz) {
 		ret = bcm2708_setup_state(spi->master, &spi->dev, &state,
-			spi->max_speed_hz, spi->chip_select, spi->mode,
+			xfer->speed_hz ? xfer->speed_hz : spi->max_speed_hz,
+			spi->chip_select, spi->mode,
 			spi->bits_per_word);
 		if (ret)
 			return ret;
